### ISSUE TITLE
[Clang] `-Wunused-variable` false negative when not all structured bindings are marked `maybe_unused`

### ISSUE THREAD

**Issue by zwuis** [llvm/llvm-project#125810]
Clang doesn't emit warning when all structured bindings are unused but not all structured bindings are marked `maybe_unused`.

```cpp
struct S {
  int a, b;
};

void test(S s) {
  auto &[a1, _] = s; // "unused variable" warning emitted expectedly
  auto &[a2, b2 [[maybe_unused]]] = s; // no warning emitted unexpectedly
  auto &[_, b3 [[maybe_unused]]] = s; // no warning emitted expectedly
}
```

Compiler Explorer: <https://godbolt.org/z/vnf9Ma3nb>

**Comment by a-tarasyuk** [comment]
https://github.com/llvm/llvm-project/blob/343bbda140d5a15cd7d7fbfc6041a7506da5cdae/clang/lib/Sema/SemaDecl.cpp#L1924-L1931

> By marking one of the member unused, we don't warn on any of the member (https://github.com/llvm/llvm-project/pull/89906/files#r1579584108)

@cor3ntin it seems this is the expected behavior, or should a warning be triggered for `a2`?

/cc @AaronBallman
